emcc.py

```diff
@@ -1041,9 +1047,10 @@ def get_last_setting_change(setting):
     if not shared.Settings.STRICT:
       # The preprocessor define EMSCRIPTEN is deprecated. Don't pass it to code
       # in strict mode. Code should use the define __EMSCRIPTEN__ instead.
-      shared.COMPILER_OPTS += ['-DEMSCRIPTEN']
+      newargs += ['-DEMSCRIPTEN']
 
-    settings_changes.append(process_libraries(libs, lib_dirs, input_files))
+    consumed = process_libraries(libs, lib_dirs, input_files)
+    link_flags = [l for l in link_flags if l[0] not in consumed]
```

Comment:
`l[0]` is the order number, maybe that should be `l[1]`?

Reply:
Yes, I'm trying to remove that flags that are not longer needed because they were consumed by process_libraries.  process_libraries returns a list of arg indexes.